test/unit_tests/braket/circuits/test_circuit.py

```diff
@@ -1012,6 +1012,376 @@ def test_as_unitary_one_gate_returns_expected_unitary(circuit, expected_unitary)
     assert np.allclose(circuit.as_unitary(), expected_unitary)
 
 
+def test_to_unitary_empty_instructions_returns_empty_array():
+    circ = Circuit()
+    circ.to_unitary() == []
+
+
+@pytest.mark.parametrize(
+    "circuit",
+    [
+        (Circuit().phaseshift(0, 0.15).apply_gate_noise(noise.Noise.BitFlip(probability=0.1))),
+        (Circuit().cnot(1, 0).apply_gate_noise(noise.Noise.TwoQubitDepolarizing(probability=0.1))),
+        (
+            Circuit()
+            .x(1)
+            .i(2)
+            .apply_gate_noise(noise.Noise.BitFlip(probability=0.1), target_qubits=[1])
+        ),
+        (
+            Circuit()
+            .x(1)
+            .i(2)
+            .apply_gate_noise(noise.Noise.BitFlip(probability=0.1), target_qubits=[2])
+        ),
+        (Circuit().x(1).i(2).apply_gate_noise(noise.Noise.BitFlip(probability=0.1))),
+        (Circuit().x(1).apply_gate_noise(noise.Noise.BitFlip(probability=0.1)).i(2)),
+        (
+            Circuit()
+            .y(1)
+            .z(2)
+            .apply_gate_noise(noise.Noise.BitFlip(probability=0.1), target_qubits=[1])
+        ),
+        (
+            Circuit()
+            .y(1)
+            .z(2)
+            .apply_gate_noise(noise.Noise.BitFlip(probability=0.1), target_qubits=[2])
+        ),
+        (Circuit().y(1).z(2).apply_gate_noise(noise.Noise.BitFlip(probability=0.1))),
+        (Circuit().y(1).apply_gate_noise(noise.Noise.BitFlip(probability=0.1)).z(2)),
+        (
+            Circuit()
+            .cphaseshift(2, 1, 0.15)
+            .si(3)
+            .apply_gate_noise(
+                noise.Noise.TwoQubitDepolarizing(probability=0.1), target_qubits=[1, 2]
+            )
+        ),
+        (
+            Circuit()
+            .cphaseshift(2, 1, 0.15)
+            .apply_gate_noise(noise.Noise.TwoQubitDepolarizing(probability=0.1))
+            .si(3)
+        ),
+    ],
+)
+@pytest.mark.xfail(raises=TypeError)
+def test_to_unitary_noise_raises_error(circuit):
+    circuit.to_unitary()
+
+
+@pytest.mark.xfail(raises=TypeError)
+def test_to_unitary_parameterized():
+    theta = FreeParameter("theta")
+    circ = Circuit().rx(angle=theta, target=0)
+    assert np.allclose(circ.to_unitary())
+
+
+def test_to_unitary_noise_not_apply_returns_expected_unitary(recwarn):
+    circuit = (
+        Circuit()
+        .cphaseshift(1, 2, 0.15)
+        .si(3)
+        .apply_gate_noise(noise.Noise.TwoQubitDepolarizing(probability=0.1), target_qubits=[1, 3])
+    )
+
+    assert len(recwarn) == 1
+    assert str(recwarn[0].message).startswith("Noise is not applied to any gate")
+
+    assert np.allclose(
+        circuit.to_unitary(),
+        np.kron(gates.CPhaseShift(0.15).to_matrix(), gates.Si().to_matrix()),
+    )
+
+
+@pytest.mark.parametrize(
+    "circuit,expected_unitary",
+    [
+        (Circuit().h(0), gates.H().to_matrix()),
```

Comment:
could we also add a test for non-contiguous qubits? 
Even just adding `Circuit().h(1), Circuit().h(2)` as test cases could work

Reply:
added.